sdks/go/pkg/beam/io/datastoreio/datastore.go

```diff
@@ -55,20 +64,25 @@ func init() {
 // datastoreio.Read(s, "project", "Item", 256, reflect.TypeOf(Item{}), itemKey)
 func Read(s beam.Scope, project, kind string, shards int, t reflect.Type, typeKey string) beam.PCollection {
 	s = s.Scope("datastore.Read")
-	return query(s, project, kind, shards, t, typeKey)
+	return query(s, project, kind, shards, t, typeKey, datastoreNewClient)
 }
 
-func query(s beam.Scope, project, kind string, shards int, t reflect.Type, typeKey string) beam.PCollection {
+func datastoreNewClient(ctx context.Context, projectID string, opts ...option.ClientOption) (clientType, error) {
+	return datastore.NewClient(ctx, projectID, opts...)
+}
+
+func query(s beam.Scope, project, kind string, shards int, t reflect.Type, typeKey string, newClient newClientFuncType) beam.PCollection {
 	imp := beam.Impulse(s)
-	ex := beam.ParDo(s, &splitQueryFn{Project: project, Kind: kind, Shards: shards}, imp)
+	ex := beam.ParDo(s, &splitQueryFn{Project: project, Kind: kind, Shards: shards, newClientFunc: newClient}, imp)
 	g := beam.GroupByKey(s, ex)
-	return beam.ParDo(s, &queryFn{Project: project, Kind: kind, Type: typeKey}, g, beam.TypeDefinition{Var: beam.XType, T: t})
+	return beam.ParDo(s, &queryFn{Project: project, Kind: kind, Type: typeKey, newClientFunc: newClient}, g, beam.TypeDefinition{Var: beam.XType, T: t})
 }
 
 type splitQueryFn struct {
-	Project string `json:"project"`
-	Kind    string `json:"kind"`
-	Shards  int    `json:"shards"`
+	Project       string `json:"project"`
+	Kind          string `json:"kind"`
+	Shards        int    `json:"shards"`
+	newClientFunc newClientFuncType
```

Comment:
Same thing here, a `Setup` method, to set up the real datastore call.

Reply:
Thanks! Comments addressed.